lib/mix/tasks/phx.gen.auth/migration.ex

```diff
@@ -12,19 +12,34 @@ defmodule Mix.Tasks.Phx.Gen.Auth.Migration do
   end
 
   defp extensions(Ecto.Adapters.Postgres) do
-    ["execute \"CREATE EXTENSION IF NOT EXISTS citext\", \"\""]
+    if case_insensitive_field_type() == :citext do
+      ["execute \"CREATE EXTENSION IF NOT EXISTS citext\", \"\""]
+    else
+      []
+    end
   end
 
   defp extensions(_), do: []
 
+  defp case_insensitive_field_type do
+    case Application.get_env(Mix.Phoenix.otp_app(), :generators, [])[:case_insensitive_field_type] do
+      type when type in [nil, :citext] -> :citext
+      custom_type -> custom_type
+    end
+  end
```

Comment:
```suggestion
  defp case_insensitive_field_type(default) do
    Application.get_env(Mix.Phoenix.otp_app(), :generators, [])[:case_insensitive_field_type] || default
  end
```

(see comment below!)